src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/filingpackage/service/FilingPackageServiceImplTest.java

```diff
@@ -24,14 +55,134 @@
+        Assertions.assertEquals(DESCRIPTION, result.get().getDocuments().get(0).getDescription());
+        Assertions.assertFalse(result.get().getDocuments().get(0).getIsAmendment());
+        Assertions.assertTrue(result.get().getDocuments().get(0).getIsSupremeCourtScheduling());
+        Assertions.assertEquals(MIME_TYPE, result.get().getDocuments().get(0).getMimeType());
+        Assertions.assertEquals(NAME, result.get().getDocuments().get(0).getName());
+        Assertions.assertEquals(BigDecimal.ONE, result.get().getDocuments().get(0).getStatutoryFeeAmount());
+    }
+
+    @Test
+    @DisplayName("Not found: missing account")
+    public void withValidRequestButMissingAccountReturnEmpty() {
+        Optional<FilingPackage> result = sut.getCSOFilingPackage(TestHelpers.CASE_2, BigDecimal.ONE);
+
+        Assertions.assertFalse(result.isPresent());
+    }
+
+
+    @Test
+    @DisplayName("Not found: no filing package")
+    public void withValidRequestButMissingPackageReturnEmpty() {
+
+        Mockito.when(efilingStatusServiceMock.findStatusByPackage(ArgumentMatchers.any())).thenReturn(Optional.empty());
+
+        Optional<FilingPackage> result = sut.getCSOFilingPackage(TestHelpers.CASE_1, BigDecimal.TEN);
+
+        Assertions.assertFalse(result.isPresent());
+    }
+
+    private AccountDetails createAccount(BigDecimal clientId) {
+        return AccountDetails.builder()
+                .fileRolePresent(true)
+                .accountId(BigDecimal.ONE)
+                .clientId(clientId)
+                .cardRegistered(true)
+                .universalId(UUID.randomUUID())
+                .internalClientNumber(null)
+                .universalId(TestHelpers.CASE_1).create();
+    }
+
+    private ca.bc.gov.open.jag.efilingcommons.submission.models.FilingPackage createFilingPackage() {
+        return new ca.bc.gov.open.jag.efilingcommons.submission.models.FilingPackage(
+                BigDecimal.ONE,
+                createCourt(),
+                Collections.singletonList(createDocument()),
+                Collections.singletonList(createParty()),
+                false,
+                "APPLICATION_CODE"
+        );
+    }
+
+    private Court createCourt() {
+        return Court.builder()
+                .agencyId(BigDecimal.ONE)
+                .classDescription(CLASS_DESCRIPTION)
+                .courtClass(COURT_CLASS)
+                .division(DIVISION)
+                .fileNumber(FILE_NUMBER)
+                .level(LEVEL)
+                .levelDescription(LEVEL_DESCRIPTION)
+                .location(LOCATION)
+                .locationDescription(LOCATION_DESCRIPTION)
+                .participatingClass(PARTICIPATING_CLASS)
+                .create();
+    }
+
+    private Party createParty() {
+        return Party.builder()
+                .firstName(FIRST_NAME)
+                .lastName(LAST_NAME)
+                .middleName(MIDDLENAME)
```

Comment:
nit: should middle name also be "MIDDLE_NAME"?

Reply:
Good call